lib/V8/v8-utils.cpp

```diff
@@ -3044,6 +3044,45 @@ static void JS_Read(v8::FunctionCallbackInfo<v8::Value> const& args) {
   TRI_V8_TRY_CATCH_END
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief was docuBlock JS_Read
+////////////////////////////////////////////////////////////////////////////////
+
+static void JS_ReadPipe(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate)
+  v8::HandleScope scope(isolate);
+
+  if (args.Length() != 1) {
+    TRI_V8_THROW_EXCEPTION_USAGE("read(pipefd)");
+  }
+
+  int64_t f = TRI_ObjectToInt64(isolate, args[0]);
+  if (f == 0) {
+    TRI_V8_THROW_TYPE_ERROR("<filename> a file pointer");
+  }
+
+  // TRI_GET_GLOBALS();
+  // V8SecurityFeature& v8security = v8g->_server.getFeature<V8SecurityFeature>();
+  // 
+  // if (!v8security.isAllowedToAccessPath(isolate, *name, FSAccessType::READ)) {
+  //   THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_FORBIDDEN,
+  //                                  std::string("not allowed to read files in this path: ") + *name);
+  // }
+
+  char content[1024];
+  size_t length = sizeof(content) - 1;
+  memset(content, 0, sizeof(content) - 1);
+  bool haveMore = TRI_ReadPointer(f, content, length);
+  if (!haveMore) {
+    length = strlen(content);
```

Comment:
is this supposed to work with binary data too? right now, it will only work with UTF-8 strings.

Reply:
should we make it more generic and return a V8 Buffer object?

Reply:
atm. it is intended for UTF8-streams (go prints json).
I had problems that we don't seem to have a way to add two buffers.